library/core/class.oauth2.php

```diff
@@ -788,13 +790,13 @@ public function requestAccessToken($code, $refresh=false) {
      */
     public function translateProfileResults($rawProfile = []) {
         $provider = $this->provider();
-        $email = val('ProfileKeyEmail', $provider, 'email');
         $translatedKeys = [
-            val('ProfileKeyEmail', $provider, 'email') => 'Email',
-            val('ProfileKeyPhoto', $provider, 'picture') => 'Photo',
-            val('ProfileKeyName', $provider, 'displayname') => 'Name',
-            val('ProfileKeyFullName', $provider, 'name') => 'FullName',
-            val('ProfileKeyUniqueID', $provider, 'user_id') => 'UniqueID'
+            $provider['ProfileKeyEmail'] ?? 'email' => 'Email',
+            $provider['ProfileKeyPhoto'] ?? 'picture' => 'Photo',
+            $provider['ProfileKeyName'] ?? 'displayname' => 'Name',
+            $provider['ProfileKeyFullName'] ?? 'name' => 'FullName',
+            $provider['ProfileKeyUniqueID'] ?? 'user_id' => 'UniqueID',
+            $provider['ProfileKeyRoles'] ?? 'roles' => 'Roles'
```

Comment:
For the sake of improving readability...

```suggestion
            ($provider['ProfileKeyEmail'] ?? 'email') => 'Email',
            ($provider['ProfileKeyPhoto'] ?? 'picture') => 'Photo',
            ($provider['ProfileKeyName'] ?? 'displayname') => 'Name',
            ($provider['ProfileKeyFullName'] ?? 'name') => 'FullName',
            ($provider['ProfileKeyUniqueID'] ?? 'user_id') => 'UniqueID',
            ($provider['ProfileKeyRoles'] ?? 'roles') => 'Roles'
```